server/backend/database/memory/housekeeping_test.go

```diff
@@ -59,14 +63,71 @@ func TestHousekeeping(t *testing.T) {
 		clientC, err := memdb.ActivateClient(ctx, project.ID, fmt.Sprintf("%s-C", t.Name()))
 		assert.NoError(t, err)
 
-		candidates, err := memdb.FindDeactivateCandidates(
+		_, candidates, err := memdb.FindDeactivateCandidates(
 			ctx,
 			10,
+			10,
+			database.DefaultProjectID,
 		)
 		assert.NoError(t, err)
 		assert.Len(t, candidates, 2)
 		assert.Contains(t, candidates, clientA)
 		assert.Contains(t, candidates, clientB)
 		assert.NotContains(t, candidates, clientC)
 	})
+
+	t.Run("housekeeping pagination test", func(t *testing.T) {
+		ctx := context.Background()
+		memdb, projects := createDBandProjects(t)
+
+		fetchSize := 4
+		lastProjectID, _, err := memdb.FindDeactivateCandidates(
+			ctx,
+			0,
+			fetchSize,
+			database.DefaultProjectID,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, projects[fetchSize-1].ID, lastProjectID)
+
+		lastProjectID, _, err = memdb.FindDeactivateCandidates(
+			ctx,
+			0,
+			fetchSize,
+			lastProjectID,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, projects[fetchSize*2-1].ID, lastProjectID)
+
+		lastProjectID, _, err = memdb.FindDeactivateCandidates(
+			ctx,
+			0,
+			fetchSize,
+			lastProjectID,
+		)
+		assert.NoError(t, err)
+		assert.Equal(t, database.DefaultProjectID, lastProjectID)
+	})
+}
+
+func createDBandProjects(t *testing.T) (*memory.DB, []*database.ProjectInfo) {
+	t.Helper()
```

Comment:
This is the first time to see `t.Helper`. When is this useful? Would it be good to add `t.Helper` to other helper functions too?

Reply:
> `t.Helper()` indicates to the Go test runner that `createDBandProjects()` helper function is a test helper. It makes it easy to report test failures at the call site of the parent function rather than the call site of `t.Error()` and `t.Errorf()`. When t.Error() is called from `createDBandProjects()` function, the Go test runner will report the filename and line number of the code which called the `createDBandProjects()` function in the output.